spec/functional/modifiers_spec.rb

```diff
@@ -173,10 +173,10 @@ def assert_keys_removed(page, *keys)
           }.to_not raise_error
         end
 
-        it "should not typecast keys that are not defined in document" do
+        it "should not typecast keys that are not defined in document and have no default typecasing" do
           expect {
             page_class.set(page.id, :colors => ['red', 'green'].to_set)
-          }.to raise_error(NoMethodError)
+          }.to raise_error(BSON::Error::UnserializableClass, /does not define its BSON serialized type/)
```

Comment:
Since bson 4.9.4, it raises BSON::Error::UnserializableClass for Set types.
https://github.com/mongodb/bson-ruby/commit/1e2a7d19004dc2383bf99685234ae820f3110054#diff-822623984add7c90fa86e969e204641428d7e498a2d943543175cbec562e62a5R159-R164

This commit follows the master branch of mongomapper/mongomapper.
https://github.com/mongomapper/mongomapper/blob/fedd68e25516703c6cf8baae7f992060af098fa3/spec/functional/modifiers_spec.rb#L176-L180